spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/core/environment/OnGcpEnvironmentConditionTests.java

```diff
@@ -52,15 +55,16 @@
 
   @Mock GcpEnvironmentProvider mockGcpEnvironmentProvider;
 
-  @Before
-  public void setUp() {
+  @BeforeEach
+  void setUp() {
     when(this.mockContext.getBeanFactory()).thenReturn(this.mockBeanFactory);
     when(this.mockBeanFactory.getBean(GcpEnvironmentProvider.class))
```

Comment:
This might be nit, but I feel like there is a slight advantage to have mock initialized with `Mockito.lenient()` here instead of having the `@MockitoSettings(strictness = Strictness.LENIENT)` annotations on each test. This way it's more clear to me which mock is optional and allowed to be not called in some of the test methods. What do you think?
```suggestion
    Mockito.lenient().when(this.mockBeanFactory.getBean(GcpEnvironmentProvider.class))
```

Reply:
I would have preferred this if all the tests had UnnecessaryStubbingException. Since only half of them have it (and the other half works fine with strict stubbing), I think it is better to not initialize mockito with lenient settings. Strict stubbing is actually preferable over lenient one as it contributes towards a cleaner and more maintainable codebase so the more we have of it, the better. Moreover, all the spring-cloud-gcp tests have mock initialized normally so this would maintain the uniformity as well. What do you think?

Reply:
Let me try removing the lenient annotation for this one. I'm working on it.